parts/k8s/addons/secrets-store-csi-driver.yaml

```diff
@@ -367,6 +367,9 @@ spec:
               mountPropagation: Bidirectional
             - name: providers-dir
               mountPath: /etc/kubernetes/secrets-store-csi-providers
+            - name: custom-environment
```

Comment:
Let's wrap this in a `{{if IsAzureStack}}` condition.

Reply:
Sorry, looks like `{{if IsAzureStackCloud}}` is what we'll need. (`IsAzureStackCloud` is the existent func in the `getAddonFuncMap` that the addons files are go-template-generated w/

Reply:
`{{if IsCustomCloudProfile}}` probably better?

Reply:
so I assume `azurestackcloud.json` is what all customCloudProfile-enabled clouds will get?

Reply:
correct

Reply:
https://github.com/Azure/aks-engine/blob/master/parts/k8s/cloud-init/masternodecustomdata.yml#L537-L543